library/std/src/net/ip.rs

```diff
@@ -1464,10 +1698,21 @@ impl Ipv6Addr {
     /// assert_eq!(Ipv6Addr::new(0xff00, 0, 0, 0, 0, 0, 0, 0).is_multicast(), true);
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0xc00a, 0x2ff).is_multicast(), false);
     /// ```
+    ///
+    /// # Stability guarantees
+    ///
+    /// This method's behavior [may by subject to changes][changes] in the
+    /// future, as new RFCs are published.
+    ///
+    /// [changes]: ../net/index.html#stability-guarantees-for-ietf-defined-behavior
     #[rustc_const_unstable(feature = "const_ipv6", issue = "76205")]
     #[stable(since = "1.7.0", feature = "ip_17")]
     pub const fn is_multicast(&self) -> bool {
-        (self.segments()[0] & 0xff00) == 0xff00
+        if let Some(v4_addr) = self.to_ipv4() {
```

Comment:
Documenting that this is a behavioral breaking change for:

- `::224.254.0.0` / `::ffff:224.254.0.0`
- `::236.168.10.65` / `::ffff:236.168.10.65`